[Feat] #31 - 홈화면 Reactor 구현

uddt-ds opened the pull request:
## 📌 관련 이슈
<!-- 관련있는 이슈 번호(#000)을 적어주세요.
  해당 pull request merge와 함께 이슈를 닫으려면 
  closed: #Issue_number를 적어주세요 -->
closed: #31

## 📌 변경 사항 및 이유
<!-- 변경한 내용과 그 이유를 적어주세요. -->
- 홈화면 Reactor를 구현하였습니다

## 📌 PR Point
<!-- 리뷰어 분들이 집중적으로 보셨으면 하는 내용을 적어주세요 -->
- 수정된 파일은 HomeReactor와 SituationDataManager, SituationDataModel입니다
- SituationDataManager와 DataModel에는 전체 카테고리가 포함되었습니다

## 📌 참고 사항
<!-- 참고할 사항이 있다면 적어주세요. -->
- Service 디렉토리 내 파일은 develop Branch를 머지하면서 생성된 파일입니다

## ✅ Checklist
<!-- 아래 내용은 확인 해주세요. -->
- [x] 주석 및 프린트문 제거 확인
- [x] 컨벤션 준수 확인

kkt6394 committed:
Merge branch 'develop' into feat/#4

kkt6394 committed:
[Feat] userDefaultsManager, GoogleAPI, NetworkManager 구현

kkt6394 committed:
Merge branch 'develop' into feat/#4

kkt6394 committed:
Merge branch 'feat/#4' into feat/#25

kkt6394 committed:
[Feat] TestViewController 생성

kkt6394 committed:
[Feat] NetworkManager, UserDefaultManager 완성

kkt6394 committed:
[Style] 오타 수정

kkt6394 committed:
[Style] 오타수정

kkt6394 committed:
[Style] 공백 삭제

kkt6394 committed:
[Style] 오타 수정

kkt6394 committed:
Merge pull request #27 from uddt-ds/feat/#25

[Feat] #25 네트워크 매니저 구현, UserDefaults 매니저 구현

uddt-ds committed:
[Feat] #31 - 홈화면 Reactor 구현

홈화면을 위한 Reactor를 구현하였습니다

* view에 바인딩하면서 테스트해보고 누락된 로직 테스트 예정

uddt-ds committed:
Merge branch 'feat/#8' into feat/#31

uddt-ds committed:
[Refactor] #31 - 홈화면 Reactor, DataModel 수정

Reactor 내부 프로퍼티로 전체 데이터를 가지고 있게끔 수정하였습니다
pushLocationView의 type을 명시적으로 작성하였습니다

홈화면의 카테고리 중 전체 카테고리가 필요함에 따라
HomeReactor,SituationDataModel, DataManager를 수정하였습니다

uddt-ds committed:
[Refactor] #31 - 홈화면 Reactor 수정

DetailView를 push할 때
keyword(검색어 예: 스시, 치킨)와 title(오늘은 월급날) 데이터를 함께 보내줘야하기 때문에,

keyword만 보내주는 단일 [String] Type에서
Section Data를 넘겨주는 방식으로 변경하였습니다

uddt-ds committed:
Merge branch 'epic/home' into feat/#31

LCH-1228 commented on EatsOkay/Home/HomeReactor.swift:
```diff
@@ -0,0 +1,75 @@
+    struct State {
+        var currentLocation: UserDeafaultsManager.UserLocation// 로케이션 초기값 필요
+        var cardSectionData: [SectionData] = []
+        @Pulse var pushLocationView: Bool? = nil
+        @Pulse var pushDetailViewWithData: SectionData? = nil
```
State에서 Optional로 타입을 명시하고 HomeReactor init시에 State에 pushLocationView와 PushDetailVIewWithData에 값이 지정되어있지 않아 자동을 nil이 들어가기에 별도의 초기값 설정을 안해줘도 될거 같습니다.

uddt-ds replied:
오..! 감사합니다!!!
확인해보겠습니다!!

LCH-1228 commented on EatsOkay/Home/HomeReactor.swift:
```diff
@@ -0,0 +1,75 @@
+class HomeReactor: Reactor {
+    var initialState: State
+
+    init(initialLocation: UserDeafaultsManager.UserLocation) {
+        self.initialState = State(currentLocation: initialLocation)
+    }
```
HomeReactor init시에 UserDefaultsManager의 UserLocation을 주입받도록 되어있는데 이전에 SA진행시 데이터는 LocationSelectReactor에서는 HomeView로 전환할때 HomeReactor에 별도 값 주입 없이 UserDefaultManger에서 불러와서 사용하도록 했던걸로 기억합니다.
혹시 해당부분은 테스트를 위해서 작성된 코드인가요?

uddt-ds replied:
제가 착오가 있어서 init에서 주입해줘야하는것으로 생각했는데
UserDefaults Manager에서 현재 저장된 장소 가져오면 될거 같습니다! 
감사합니다!!

hyemin815 reviewed (approved):
Reactor 구조가 잘 짜여져있고, 위치 정보는 viewWillAppear에 카드 섹션은 viewDidLoad에 동작하도록 구분하여 작성하신 부분이 인상 깊습니다!

LCH-1228 commented on EatsOkay/Home/HomeReactor.swift:
```diff
@@ -0,0 +1,75 @@
+        case .setCardSection(let sectionData):
+            newState.cardSectionData = sectionData
+        case .requestLocationView:
+            newState.pushLocationView = true
```
Bool값의 활용이 없다면 Optional Void를 사용해도 될거 같습니다.

uddt-ds replied:
수정하겠습니다~~

heopill commented on EatsOkay/Home/HomeReactor.swift:
```diff
@@ -0,0 +1,75 @@
+    struct State {
+        var currentLocation: UserDeafaultsManager.UserLocation// 로케이션 초기값 필요
+        var cardSectionData: [SectionData] = []
+        @Pulse var pushLocationView: Bool? = nil
```
여기서 @Pulse 는 어떤 걸 의미하나요?

uddt-ds replied:
@Pulse는 ReactorKit에서 값이 바뀔 때 시그널을 주려고 사용하는데요!

값 자체는 필요 없고, '값이 바뀌었다'는 것만 인지시킬 때 사용합니다
여기서 저의 경우에는 view에서 push를 여러번 할 필요가 없고,
이벤트 발생 시 딱 1번만 하면 되기 때문에 Pulse를 사용했습니다!

heopill reviewed (approved):
고생하셨습니다 :)

uddt-ds committed:
[Refactor] #31 - 홈화면 Reactor 수정

pushLocationView의 Type을 수정하였습니다.
pushDetailViewWithData의 Type을 수정하였습니다.

uddt-ds committed:
[Refactor] #31 - 홈화면 Reactor init 코드 수정

- init 시점에서 값의 주입이 필요하지 않아 코드를 수정하였습니다.
- loadCurrentLocation에서 UserLocation의 Address만 필요하여, loadCurrentLocation을 String Type으로 변경하였습니다

LCH-1228 commented on EatsOkay/Home/HomeReactor.swift:
```diff
@@ -0,0 +1,76 @@
+            newState.cardSectionData = sectionData
+        case .requestLocationView:
+            newState.pushLocationView = ()
+        case .pushDetailView(data: let data):
```
case .pushDetailView(let data): 로 사용하셔도 무방할거 같습니다.

uddt-ds replied:
수정하겠습니다!!

uddt-ds committed:
[Refactor] #31 - 홈화면 Reactor 오타 수정

- Void Type을 명시적으로 변경하였습니다.
- data: let data 오타를 수정하였습니다.

LCH-1228 commented on EatsOkay/Home/HomeReactor.swift:
```diff
@@ -0,0 +1,76 @@
+        case .setCardSection(let sectionData):
+            newState.cardSectionData = sectionData
+        case .requestLocationView:
+            newState.pushLocationView = ()
```
Void()로 명시해주세요

uddt-ds replied:
반영하였습니다!!
